kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseSiteService.java

```diff
@@ -2939,6 +3017,47 @@ protected void disableAuthorizationGroup(String ref)
 		 */
 		public void saveInfo(String siteId, String description, String infoUrl);
 
+		/**
+		 * Unpublish the site by simply unsetting the PUBLISHED flag
+		 * @param siteId
+		 *        The ID of the site to unpublish
+		 * @param modifedBy
+		 *        User who is unpublishing the site (as a userID)
+		 * @param modifiedOn
+		 *        Time that the site is unpublished
+		 */
+		public void unpublish(String siteId, String modifiedBy, Time modifiedOn);
+
+		/**
+		 * Unpublish the sites by simply unsetting the PUBLISHED flag
+		 * @param siteIds
+		 *        The site to unpublish
+		 * @param modifedBy
+		 *        User who is unpublishing the site (as a userID)
+		 * @param modifiedOn
+		 *        Time that the site is unpublished
+		 */
+		public void unpublish(List<String> siteIds, String modifiedBy, Time modifiedOn);
+
+		/**
+		 * Writes site properties
+		 */
+		public void writeProperties(Entity r, ResourceProperties props);
+
+		/**
+		 * Saves a site property for the site with the specified ID using the specified name-value pair.
+		 * NB: Inserts only; doesn't do any duplicate checking. Vulnerable to unique constraint violations
+		 * Use this only when making very minimal changes in performance critical tasks.
+		 */
+		public void writeProperty(String siteId, String propertyName, String propertyValue);
```

Comment:
if you wrote this method signature using varargs would only need one method vs 2:
public void writeProperty(String propertyName, String propertyValue, String... siteId);

Reply:
I agree that it would be better to only have one method. However I think it would be more appropriate for the method to take a `List<String>` rather than using varargs. The main reason being here that the list of site IDs for some situtaions could be very long, and calling a method with say 100 arguments can be cumbersome and hard to read.

Using a `List<String>` as the parameter accomplishes the same goal of reducing the API, and we have Collections.singletonList() for passing a single object list.